test/raft/raft_address_map_test.cc

```diff
@@ -52,79 +52,60 @@ SEASTAR_THREAD_TEST_CASE(test_raft_address_map_operations) {
     using seastar::manual_clock;
 
     {
-        // Set regular entry + set_expiring_flag works as expected
-        sharded<raft_address_map<manual_clock>> m_svc;
+        sharded<raft_address_map_t<manual_clock>> m_svc;
         m_svc.start().get();
         auto stop_map = defer([&m_svc] { m_svc.stop().get(); });
         auto& m = m_svc.local();
 
-        m.set(id1, addr1, false);
-        {
-            const auto found = m.find(id1);
-            BOOST_CHECK(!!found && *found == addr1);
-        }
-        {
-            const auto found = m.set_expiring_flag(id1);
-            BOOST_CHECK(!!found && *found == addr1);
-        }
+        m.add_or_update_entry(id1, addr1);
+        m.set_nonexpiring(id1);
+        // Check that regular entries don't expire
         manual_clock::advance(expiration_time);
-        BOOST_CHECK(!m.find(id1));
-    }
-    {
-        // Set expiring entry + set_expiring_flag works as expected
-        sharded<raft_address_map<manual_clock>> m_svc;
-        m_svc.start().get();
-        auto stop_map = defer([&m_svc] { m_svc.stop().get(); });
-        auto& m = m_svc.local();
-
-        m.set(id1, addr1, true);
         {
             const auto found = m.find(id1);
             BOOST_CHECK(!!found && *found == addr1);
         }
-        {
-            const auto found = m.set_expiring_flag(id1);
-            BOOST_CHECK(!!found && *found == addr1);
-        }
+        m.set_expiring(id1);
         manual_clock::advance(expiration_time);
         BOOST_CHECK(!m.find(id1));
     }
     {
-        // Set expiring + wait for expiration works as expected
-        sharded<raft_address_map<manual_clock>> m_svc;
+        // m.add_or_update_entry() adds an expiring entry
+        sharded<raft_address_map_t<manual_clock>> m_svc;
         m_svc.start().get();
         auto stop_map = defer([&m_svc] { m_svc.stop().get(); });
         auto& m = m_svc.local();
 
-        m.set(id1, addr1, true);
+        m.add_or_update_entry(id1, addr1);
         auto found = m.find(id1);
         BOOST_CHECK(!!found && *found == addr1);
         // The entry stay in the cache for 1 hour
         manual_clock::advance(expiration_time);
         BOOST_CHECK(!m.find(id1));
     }
     {
-        // Check that regular entries don't expire
-        sharded<raft_address_map<manual_clock>> m_svc;
+        // To mark an entry non-expiring, call set_nonexpiring().
+        sharded<raft_address_map_t<manual_clock>> m_svc;
         m_svc.start().get();
         auto stop_map = defer([&m_svc] { m_svc.stop().get(); });
         auto& m = m_svc.local();
 
-        m.set(id1, addr1, false);
+        m.add_or_update_entry(id1, addr1);
+        m.set_nonexpiring(id1);
```

Comment:
This is redundant, it's exactly the same as the first test now.

Reply:
Removed.